### PR TITLE
ofc: update to 2023.02.06, fix build on systems where it fails

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update, fix the build on systems where it fails currently.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.1.1
Xcode 15.0.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: lang/ofc/Portfile [hunk 1]
```diff
@@ -9,28 +9,39 @@ PortGroup           makefile 1.0
...
+        if {${os.platform} eq "darwin" && ${os.major} > 20} {
+            # unformat.c: error: variable 'row' set but not used [-Werror,-Wunused-but-set-variable]
+            # https://github.com/CodethinkLabs/ofc/issues/49
+            reinplace "s|-Werror|-Werror -Wno-error=unused-but-set-variable|g" ${worksrcpath}/Makefile
+        } elseif {${os.platform} eq "darwin" && ${os.major} == 19} {
+            # file.c: error: declaration does not declare anything [-Werror,-Wmissing-declarations]
+            # https://github.com/CodethinkLabs/ofc/issues/46
+            reinplace "s|-Werror|-Werror -Wno-error=missing-declarations|g" ${worksrcpath}/Makefile
+        }
+    }
```
Comment (reneeotten) [inline lang/ofc/Portfile:43]: why not remove `-Werror` regardless of the OS versions - we normally don't build with that flag anyway. I don't think there is a reason to have these conditionals here.

Comment (barracuda156) [reply lang/ofc/Portfile:43]: @reneeotten I would rather keep it, but won’t insist: if you prefer dropping -Werror unconditionally, I am fine with that either.

Comment (reneeotten) [reply lang/ofc/Portfile:43]: I've seen Josh recently replacing `-Werror` with `-Wextra` in a few ports, so perhaps doing that unconditionally would be a suitable solution?

Comment (barracuda156) [reply lang/ofc/Portfile]: @reneeotten Since I cannot test anything on older OS with GCC at the moment, let me rather just remove `-Werror` here.

Comment (pmetzger) [reply lang/ofc/Portfile]: FWIW, I think -Werror was appropriate but it doesn't matter much. (-Wextra is like -Werror but "harder").

**Pull request merged**